OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,475 @@
+                }
+                _lastPressed = null;
+            }
+        }
+
+        private bool IsPressed_Combo(IModInputCombination combination)
+        {
+            UpdateCombo();
+            return _lastPressed == combination;
+        }
+
+        private bool IsNewlyPressed_Combo(IModInputCombination combination, bool keep = false)
+        {
+            UpdateCombo();
+            return _lastPressed == combination && combination.IsFirst(keep);
+        }
+
+        private bool WasTapped_Combo(IModInputCombination combination)
+        {
+            UpdateCombo();
+            return combination != _lastPressed && (Time.realtimeSinceStartup - combination.GetLastPressedMoment() < _tapKeep) && (combination.GetPressDuration() < _tapDuration);
+        }
+
+        private bool IsNewlyReleased_Combo(IModInputCombination combination, bool keep = false)
+        {
+            UpdateCombo();
+            return _lastPressed != combination && combination.IsFirst(keep) && (Time.realtimeSinceStartup - combination.GetLastPressedMoment() < _tapKeep);
+        }
+
+        private bool IsPressed_Single(IModInputCombination combination)
+        {
+            if (Time.realtimeSinceStartup - _lastSingleUpdate > _updateCooldown)
+            {
+                _lastSingleUpdate = Time.realtimeSinceStartup;
+                var toRemove = new List<IModInputCombination>();
+                foreach (IModInputCombination combo in _singlePressed)
+                    if (!IsPressed(combo))
+                    {
+                        ((ModInputCombination)combo).SetPressed(false);
+                        toRemove.Add(combo);
+                    }
+                foreach (IModInputCombination combo in toRemove)
+                    _singlePressed.Remove(combo);
+            }
+            List<KeyCode> keys = ((ModInputCombination)combination).GetSingles();
+            foreach (KeyCode key in keys)
+            {
+                if (UnityEngine.Input.GetKey(key) && (!ShouldIgnore(key) || _singlePressed.Contains(combination)))
+                {
+                    _singlePressed.Add(combination);
+                    _timeout[(int)key] = Time.realtimeSinceStartup;
+                    ((ModInputCombination)combination).SetPressed();
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private bool IsNewlyPressed_Single(IModInputCombination combination, bool keep = false)
+        {
+            return IsPressed_Single(combination) && combination.IsFirst(keep);
+        }
+
+        private bool WasTapped_Single(IModInputCombination combination)
+        {
+            return (!IsPressed_Single(combination)) && ((ModInputCombination)combination).IsRelevant(_tapKeep)
+                && (combination.GetPressDuration() < _tapDuration);
+        }
+
+        private bool IsNewlyReleased_Single(IModInputCombination combination, bool keep = false)
```

Comment:
Also not used

Reply:
yeah, probably should get rid of _Single versions